src/App.tsx

```diff
@@ -1,39 +1,63 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Components/Clock';
 
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+export type State = {
+  hasClock: boolean
+  clockName: string,
+};
 
-        <span className="Clock__time">
-          {date.toLocaleTimeString()}
-        </span>
+class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  newName = 0;
+
+  componentDidMount() {
+    this.newName = window.setInterval(this.nameChanger, 3300);
+    document.addEventListener('click', (event: MouseEvent) => {
+      if (event.button === 0) {
+        this.setState({ hasClock: true });
+        this.newName = window.setInterval(this.nameChanger, 3300);
```

Comment:
you'll add a new interval everytime user presses a button, that'll lead to a lot of intervals, remove it
```suggestion
```
![image](https://user-images.githubusercontent.com/37161315/179466587-0579901d-4153-45a4-b0a0-0fbd99d4e72b.png)

Reply:
Or add some type of ceck if you should add the `setInterval` again